tools/c7n_azure/c7n_azure/resources/appserviceplan.py

```diff
@@ -29,3 +34,66 @@ class resource_type(ArmResourceManager.resource_type):
             'resourceGroup',
             'kind'
         )
+
+    @staticmethod
+    def register(registry, _):
+        # Additional filters/actions registered for this resource type
+        AppServicePlan.filter_registry.register("offhour", AzureOffHour)
+        AppServicePlan.filter_registry.register("onhour", AzureOnHour)
+
+
+resources.subscribe(resources.EVENT_FINAL, AppServicePlan.register)
+
+
+@AppServicePlan.action_registry.register('resize-plan')
+class ResizePlan(AzureBaseAction):
+    """Resize App Service Plans
+
+        .. code-block:: yaml
+
+          policies:
+            - name: azure-resize-plan
+              resource: azure.appserviceplan
+              actions:
+               - type: resize-plan
+                 size: F1 # F1, D1, S1, S2, S3, P1, P2, P3
+                 count: 1
+    """
+
+    schema = utils.type_schema(
+        'resize-plan',
+        **{
+            'size': {'type': 'string', 'enum': ['F1', 'D1', 'S1', 'S2', 'S3', 'P1', 'P2', 'P3']},
+            'count': {'type': 'integer'}
+        }
+    )
+
+    def process_resource_set(self, plans):
+        client = self.manager.get_client()  # type: azure.mgmt.web.WebSiteManagementClient
+
+        for plan in plans:
+            model = models.AppServicePlan(location=plan['location'])
+
+            if 'size' in self.data:
+                size = self.data.get('size')
+                model.sku = models.SkuDescription()
+                model.sku.tier = ResizePlan.lookup_tier(size)
+                model.sku.capacity = size[1]
```

Comment:
Isn't size just a string? What is size[1] vs size[0]?

Reply:
Yeah it is a little silly and undocumented. I am simplifying it for the user because we actually have to fill out several fields that are various combinations of the first and second character of the two-character SKU name.  So that is just me indexing into either the letter or the number.  The schema enforces a fixed set of strings they can provide.

Reply:
Ah, I get it now. Interesting!